app/models/order.rb

```diff
@@ -0,0 +1,50 @@
+class Order < ActiveRecord::Base
+  has_many :order_items
+  belongs_to :user
+
+  validates :order_state, presence: true
+  validates :name, presence: true, on: :update
+  validates :email, presence: true, on: :update
+  validates :address, presence: true, on: :update
+  validates :city, presence: true, on: :update
+  validates :state, presence: true, on: :update
+  validates :zip, presence: true, on: :update
+  validates :card_name, presence: true, on: :update
+  validates :last_four_cc, presence: true, on: :update
+  validates :cc_cvv, presence: true, on: :update
+  validates :billing_zip, presence: true, on: :update
+  validates :cc_exp_month, presence: true, on: :update
+  validates :cc_exp_year, presence: true, on: :update
+
+  def already_ordered?(product, order_id)
+    item = OrderItem.find_by_product_id_and_order_id( product.id, order_id)
+
+    if item.nil?
+      return false
+    else
+      return item
+    end
+  end
+
+  def add_product(product, quantity, order_id)
+    item = already_ordered?(product, order_id)
+
+    if item
+      item.quantity += quantity.to_f
+      item.save
+    else
+      OrderItem.create(product_id: product.id, order_id: order_id, quantity: quantity)
+    end
+  end
```

Comment:
Both of the above methods are a little confusing and seem intertwined in a way that makes it a good idea to pause and consider whether this is the right place for these methods.

Whatever your choice with the infrastructure, you are calling them on an order object so you do not need to explicitly include order_id. One would presume that would just be the id of the order you are calling this method on.